datastream-server/src/main/java/com/linkedin/datastream/server/DatastreamTaskImpl.java

```diff
@@ -369,4 +453,25 @@ public void updateCheckpoint(int partition, String checkpoint) {
     LOG.debug("Update checkpoint called for partition {} and checkpoint {}", partition, checkpoint);
     _checkpoints.put(partition, checkpoint);
   }
+
+  // required for json deserialization
+  public List<String> getDependencies() {
+    return _dependencies;
+  }
+
+  /**
+   * Add a precedent task to this task
+   */
+  public void addDependency(String taskName) {
+    _dependencies.add(taskName);
+  }
+
+  /**
+   * return task version to decide if it uses partition (v1) or partition v2
+   */
+  @JsonIgnore
+  public int getTaskVersion() {
```

Comment:
I am not sure whether we need this. If we want this, we need to make it first class. @gauravrkulkarni  What is the usecase for this method?

Reply:
@gauravrkulkarni, I will probably to remove this as partitionv2 should always contains the v1 data (if it's populated). As a result, we can always check v2 one without knowing the task version.